### ISSUE TITLE
Xbox 360 Controller not recognized in Firefox (Mac)

### ISSUE THREAD

**Issue by sjml** [godotengine/godot#32831]
**Godot version:** 3.1.1

**OS/device including version:** macOS 10.14.6; Firefox 69.0.3

**Issue description:** In Chrome and the Mac build, I'm able to use an Xbox 360 Controller, which presents to the engine as "Xbox 360 Wired Controller." However, this doesn't get recognized in Firefox, perhaps because the name shows up as "45e-28e-Xbox 360 Wired Controller." I'm not sure where that extra string at the beginning is coming from, but it seems like something Godot should maybe anticipate and be able to handle? This might only be an issue on the Mac, maybe something with the 360Controller.kext that most folks use to get it running; I don't have access to my Windows machine at the moment so can't confirm. 

**Steps to reproduce:**
1. Run on a Mac using the [360Controller](https://github.com/360Controller/360Controller/) kernel extension. (As I said, may be the case elsewhere; I just don't want to make overly broad claims.)
2. Load the linked project from below and export to HTML5. 
3. Start a server to run the exported HTML.
4. Load the page in Firefox, making sure the console is visible. 
5. Move a stick or press a button on an attached 360 Controller (since a webpage won't treat it as connected until it's interacted with.)
6. Note the console output: 
```
Joystick 0 connected
name: 45e-28e-Xbox 360 Wired Controller
Device not recognized.
```
7. Run the same project from the editor. 
8. Note the console output:
```
Joystick 0 connected
name: X360 Controller
Recognized joystick
X360 Controller connected
```
9. Load the page in Chrome, with console visible. Interact with the controller.
10. Note the console output:
```
Joystick 0 connected
name:  Default Mapping
Recognized joystick
Default Mapping connected
```

**Minimal reproduction project:**
[sjml/Godot-Firefox-ControllerBug](https://github.com/sjml/Godot-Firefox-ControllerBug)

**Comment by bruvzg** [comment]
Might be related to missing bindings (incomplete SDL binding support, see #8368). Is it not working completely or not detected as XBox controller (sending wrong button codes)?

**Comment by sjml** [comment]
> Might be related to missing bindings (incomplete SDL binding support, see #8368). Is it not working completely or not detected as XBox controller (sending wrong button codes)?

Yeah, if you read the axes and buttons directly, the values look right. Is it weird that it's getting a mangled controller name like that, though? I'm guessing that's why it's ending up unrecognized.

**Comment by Anutrix** [comment]
The id is not consistent across browsers. For example, it’s `45e-28e-Wireless 360 Controller` in Mozilla Firefox, which is different from what Google Chrome provides(`Xbox 360 Controller (XInput STANDARD GAMEPAD))`.
Src: https://www.smashingmagazine.com/2015/11/gamepad-api-in-web-games/

**Comment by ndarilek** [comment]
I don't think this is limited to the Mac. I'm under Linux, and can't get my Xbox 1 controller working at all under Firefox. Haven't tried Chrome yet but will once I've worked out some other issues.

I'm using [this site](https://html5gamepad.com/) to test my gamepad. It does appear under both Firefox and Chromium, but as previous commenters have noted, it does have different names. Firefox names it "045e-02ea-Microsoft X-Box One S pad" while Chromium calls it something that appears a bit more human-readable (I.e. no leading hex.)

Just a hunch not having looked at the code. Is something checking for exact matches of gamepad names to models? If so, would anything break if the matches were containment-based? I.e. if the string "Microsoft X-Box One S pad" was checked for, then the leading product/vendor IDs wouldn't matter.)

I'll run my game in Chrome a bit later to see if that indeed works.

**Comment by ndarilek** [comment]
Confirming that, under Linux, Chromium works without issue. So this isn't Mac-specific.

**Comment by Nintorch** [comment]
Hello, may I ask if this is still an issue in Godot 4.7 and newer versions of Firefox?
